server/portal/apps/workspace/api/views.py

```diff
@@ -263,10 +263,28 @@ def post(self, request, *args, **kwargs):
         job_id = job_post.get('job_id')
         job_action = job_post.get('action')
 
-        # cancel job / stop job
         if job_id and job_action:
-            METRICS.info("user:{} is canceling/stopping job id:{}".format(request.user.username, job_id))
+            # resubmit job
+            if job_action == 'resubmit':
+                METRICS.info("user:{} is resubmitting job id:{}".format(request.user.username, job_id))
+            # cancel job / stop job
+            else:
+                METRICS.info("user:{} is canceling/stopping job id:{}".format(request.user.username, job_id))
+
             data = agave.jobs.manage(jobId=job_id, body={"action": job_action})
+
+            if job_action == 'resubmit':
+                if "id" in data:
+                    # Remove previous job fom job list
+                    prev_job = JobSubmission.objects.get(jobId=job_id)
+                    prev_job.delete()
```

Comment:
is this from CEPv1.  i wonder why they would delete the previous job.  🤔   it would be fine for it to still exist, right?  not certain just thinking "out loud"

Reply:
@nathanfranklin Nope.. this was one of my questions.
I wanted to show that the same job (in the queue) was re-running.
Would it make more sense to keep the previous job around?

Reply:
Yeah, without good reasoning I think we should definitely preserve the previous job.

Reply:
> I wanted to show that the same job (in the queue) was re-running.

Because they would both have the old job and new job are going to have the same name, right?   yeah, i think i understand.  🤔  but then if we deleted the previous job, then users would no longer be able to get to the old jobs archived data (i.e View in Data Files button). I think 🤔